cmake/ProjectWasm3.cmake

```diff
@@ -0,0 +1,48 @@
+if(ProjectWasm3Included)
+    return()
+endif()
+set(ProjectWasm3Included TRUE)
+
+include(ExternalProject)
+
+set(prefix ${CMAKE_BINARY_DIR}/_deps)
+set(source_dir ${prefix}/src/wasm3)
+set(binary_dir ${prefix}/src/wasm3-build)
+set(include_dir ${source_dir})
+set(wasm3_library ${binary_dir}/source/${CMAKE_STATIC_LIBRARY_PREFIX}m3${CMAKE_STATIC_LIBRARY_SUFFIX})
+
+# Should use a release, like https://github.com/wasm3/wasm3/archive/v0.4.6.tar.gz, once changes are upstreamed.
+
+ExternalProject_Add(wasm3
+    EXCLUDE_FROM_ALL 1
+    PREFIX ${prefix}
+    DOWNLOAD_NAME wasm3-ext-api.tar.gz
+    DOWNLOAD_DIR ${prefix}/downloads
+    SOURCE_DIR ${source_dir}
+    BINARY_DIR ${binary_dir}
+    URL https://github.com/axic/wasm3/archive/ext-api.tar.gz
```

Comment:
This needed API changes to work. Found here: https://github.com/axic/wasm3/tree/ext-api

Reply:
Apparently there is a C++ API: https://github.com/wasm3/wasm3/blob/master/platforms/cpp/wasm3_cpp/include/wasm3_cpp.h

Could try using it.

Reply:
Never mind, it is not exposing memory.